server/controller/src/main/java/ai/starwhale/mlops/domain/filestorage/FileStorageService.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2022 Starwhale, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.starwhale.mlops.domain.filestorage;
+
+import static org.springframework.http.MediaType.APPLICATION_OCTET_STREAM_VALUE;
+
+import ai.starwhale.mlops.exception.SwProcessException;
+import ai.starwhale.mlops.exception.SwValidationException;
+import ai.starwhale.mlops.storage.StorageAccessService;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.convert.DurationStyle;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+
+
+@Slf4j
+@Service
+public class FileStorageService {
+    private final StorageAccessService storageAccessService;
+    private final String dataRootPath;
+    private final Long urlExpirationTimeMillis;
+
+    public FileStorageService(StorageAccessService storageAccessService,
+                              @Value("${sw.filestorage.data-root-path:files}") String dataRootPath,
+                              @Value("${sw.filestorage.url-expiration-time:24h}") String urlExpirationTime) {
+        this.storageAccessService = storageAccessService;
+        if (dataRootPath.endsWith("/")) {
+            this.dataRootPath = dataRootPath;
+        } else {
+            this.dataRootPath = dataRootPath + "/";
+        }
```

Comment:
Check if the root path is short enough to make sure that the `generatePathPrefix` less than 255?

Reply:
ok